app/assets/javascripts/discourse/app/lib/user-tips.js

```diff
@@ -3,111 +3,179 @@ import { iconHTML } from "discourse-common/lib/icon-library";
 import I18n from "I18n";
 import { escape } from "pretty-text/sanitizer";
 import tippy from "tippy.js";
+import isElementInViewport from "discourse/lib/is-element-in-viewport";
 
-const instances = {};
-const queue = [];
+const TIPPY_DELAY = 500;
+
+const instancesMap = {};
+window.instancesMap = instancesMap;
+
+function destroyInstance(instance) {
+  if (instance.showTimeout) {
+    clearTimeout(instance.showTimeout);
+    instance.showTimeout = null;
+  }
+
+  if (instance.destroyTimeout) {
+    clearTimeout(instance.destroyTimeout);
+    instance.destroyTimeout = null;
+  }
+
+  instance.destroy();
+}
+
+function cancelDestroyInstance(instance) {
+  if (instance.destroyTimeout) {
+    clearTimeout(instance.destroyTimeout);
+    instance.destroyTimeout = null;
+  }
+}
+
+function showInstance(instance) {
+  if (isTesting()) {
+    instance.show();
+  } else if (!instance.showTimeout) {
+    instance.showTimeout = setTimeout(() => {
+      instance.showTimeout = undefined;
```

Comment:
```suggestion
      instance.showTimeout = null;
```

It's a nitpick, but to me undefined is coming from "machine", and null is set by developer